### PR TITLE
Add material science domain

### PR DESCRIPTION
**Pull request by TungstnBallon**
This based on the work done [here](https://github.com/jvalue/SWC2-JValue-OCDE-Case1/tree/a0cce5e253d1674d31cf22c1484970fb0bd2eea3/models/14946186/shared)

related: #569

I didn't mark this pr as closing the above issue, because one of the user acceptance criteria is that this domain is used at  https://github.com/jvalue/SWC2-JValue-OCDE-Case1, which can only be fulfilled after this is merged

### CODE REVIEW

File: libs/language-server/src/stdlib/domain/material-science/MatSciDoiStandardizer.jv [hunk 1]
```diff
@@ -0,0 +1,77 @@
+// Transformer to standardize DOI references and remove dx.doi.org/
```
Comment (georg-schwarz) [inline libs/language-server/src/stdlib/domain/material-science/MatSciDoiStandardizer.jv:1]: No real review from my side, but I noticed that you use the "simple" `//` comments here.

If you use `/** ... */` comments, the documentation can be picked up by Langium, so we can show it when hovering over a reference or on code completion :)

Comment (TungstnBallon) [reply libs/language-server/src/stdlib/domain/material-science/MatSciDoiStandardizer.jv]: Done, thanks :)

File: libs/language-server/src/stdlib/domain/material-science/MatSciDoiStandardizer.jv
Comment (rhazn) [inline libs/language-server/src/stdlib/domain/material-science/MatSciDoiStandardizer.jv:1]: Would prefer this to be named just `DoiStandardizer.jv`.

File: libs/language-server/src/stdlib/domain/material-science/MatSciDoiStandardizer.jv [hunk 2]
```diff
@@ -0,0 +1,82 @@
+// SPDX-FileCopyrightText: 2024 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+/** Transformer to standardize DOI references and remove dx.doi.org */
+publish transform RemoveDxDotDoiDotOrg {
+	from valueWithBraces oftype text;
```
Comment (rhazn) [inline libs/language-server/src/stdlib/domain/material-science/MatSciDoiStandardizer.jv:7]: Input/output names in all transforms seem wrong (probably c&p)? Should match what the transform actually does, e.g. `from unstandardizedValue`, `to standardizedValue`.

File: libs/language-server/src/stdlib/domain/material-science/MatSciDoiStandardizer.jv [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+/** Transformer to standardize DOI references and remove https://doi.org/ */
+publish transform RemoveHttpsDoiDotOrg {
+	from valueWithBraces oftype text;
+	to valueWithoutBraces oftype text;
+
+	valueWithoutBraces: valueWithBraces replace /^https:\/\/doi\.org\/\b/ with "";
+}
+
+/** Forces all values in a column to follow the doi format. Removes prefixes like "http://" */
```
Comment (rhazn) [inline libs/language-server/src/stdlib/domain/material-science/MatSciDoiStandardizer.jv:37]: Please extend the documentation a bit, it is really important for this to be good in the standard library. For the DOIStandardizer that would mean some inputs and how they would look standardized (e.g. http://doi.org/123 -> 123 etc). I hope you can look at the GTFS blocks in the transport extension for examples of docs.

File: libs/language-server/src/stdlib/domain/material-science/MatSciDoiStandardizer.jv [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+publish composite blocktype DOIStandardizer {
+	input UnstandardizedDOI oftype Table;
+	output StandardizedDOI oftype Table;
+
+	property doiColumn oftype text;
+
+	UnstandardizedDOI
+		-> DOIStandardizerStep1
```
Comment (rhazn) [inline libs/language-server/src/stdlib/domain/material-science/MatSciDoiStandardizer.jv:45]: Naming should allow me to infer what this step does, not just be numbered.

File: libs/language-server/src/stdlib/domain/material-science/MatSciValueTypes.jv [hunk 5]
```diff
@@ -0,0 +1,52 @@
+// SPDX-FileCopyrightText: 2024 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+constraint DOIFormat on text: value matches /\b10\.\d{4}\/[^\s]+\b/;
+/** DOI Format has been constrained by a standard pattern, eg: 10.1007/xxxx */
```
Comment (rhazn) [inline libs/language-server/src/stdlib/domain/material-science/MatSciValueTypes.jv:6]: Reference here that the `DOIStandardizer` block allows you to bring dois in a format that matches this.

File: libs/language-server/src/stdlib/domain/material-science/MatSciValueTypes.jv
Comment (rhazn) [inline libs/language-server/src/stdlib/domain/material-science/MatSciValueTypes.jv:1]: I'd prefer a longer, full name like `MaterialScienceValueTypes.jv`.

**Review by rhazn — CHANGES_REQUESTED** [review]
Nice work, but some naming and docs issues.

**Comment by rhazn** [comment]
@georg-schwarz @joluj I assume this should be fine, but to make sure: Are you okay with us moving the common value types and blocks we arrived at from SWC into a basic material science domain extension for Jayvee?

**Review by rhazn — APPROVED** [review]

**Comment by rhazn** [comment]
I think in this repository you should be able to merge yourself after approvals, can you try @TungstnBallon ?

**Comment by TungstnBallon** [comment]
> I think in this repository you should be able to merge yourself after approvals, can you try @TungstnBallon ?

Yes.

**Pull request merged**